docs/configure

```diff
@@ -647,33 +647,32 @@ virtual_forward:
 
 virtual_domains:
   driver = redirect
+  address_data = ${lookup mysql{\
+        select smtp, users.sa_tag*10 AS sa_tag, users.on_spamassassin AS on_spamassassin \
+        from users,domains \
+        where localpart = '${quote_mysql:$local_part}' \
+                and domain = '${quote_mysql:$domain}' \
+                and domains.enabled = '1' \
+                and users.enabled = '1' \
+                and users.domain_id = domains.domain_id}{$value}fail}
   allow_fail
-  data = ${lookup mysql{select smtp from users,domains \
-  		where localpart = '${quote_mysql:$local_part}' \
-		and domain = '${quote_mysql:$domain}' \
-		and domains.enabled = '1' \
-		and users.enabled = '1' \
-		and users.domain_id = domains.domain_id}}
-  headers_add = ${if >{$spam_score_int}{${lookup mysql{select users.sa_tag * 10 from users,domains \
-  		where localpart = '${quote_mysql:$local_part}' \
-		and domain = '${quote_mysql:$domain}' \
-		and users.on_spamassassin = '1' \
-		and users.domain_id=domains.domain_id }{$value}fail}} {X-Spam-Flag: YES\n}{} }
-  # You may have to use single colons in headers_remove rule, or comment it out altogether
-  # if using Exim v. 4.85 or older. See https://github.com/avleen/vexim2/issues/21 and
-  # check out the links pasted in that issue for more info.
+  data = ${extract{smtp}{$address_data}}
+  headers_add = ${if and { \
+                    {>{$spam_score_int}{${extract{sa_tag}{$address_data}}}} \
+                    {eq{1}{${extract{on_spamassassin}{$address_data}}}} \
+                    } {X-Spam-Flag: YES\n}{} }
+  headers_remove = ${if or { { <{$spam_score_int}{1} } \
+                            { and {{<{$spam_score_int}{${extract{sa_tag}{$address_data}}} } \
+                                   {eq {1}{${extract{on_spamassassin}{$address_data}}}} \
+                                  } \
+                            } { eq {0}{${extract{on_spamassassin}{$address_data}}}} \
```

Comment:
That's correct, but I doubt you would want to filter your email by spam headers if you disable spam processing for your account. Either you care about these haders and you ask the server to add them, or you just don't care about them.
Then also, `X-Spam-Status` could as well be set to `NO` by the other server, and `X-Spam-Score` could be set to `1` or less. These headers seem quite useless when passed by the other server, but I'm not sure if that means we should remove them.

On the other hand, since it was this way before the patch, I'll merge it in anyway, and we can leave this discussion for later.